### DISCUSSION TITLE
createSignedUrl from database

### DISCUSSION THREAD

**Discussion by megacherry** [supabase/supabase#3060] [Questions] [answered]
Hi folks,

do somebody know if it's possible to create a signedUrl from database?

Thanks in advance.

**Comment by dshukertjr** [comment, marked as answer]
Hi @megacherry 

Could you clarify what you are trying to achieve here? You have your files stored in Supabase storage, and as well as a path to the object, and you want to create a signed URL from the object path?

**Reply by dshukertjr** [reply]
@megacherry 

If you have the path to the storage object, then it is pretty straight forward. You can do something like this to get the signed URL:

```typescript
const pathToObject = ''; // Get your path to the object from DB
const { signedURL, error } = await supabase
  .storage
  .from('avatars')
  .createSignedUrl(pathToObject, 60)
```

[createSignedURL reference](https://supabase.io/docs/reference/javascript/storage-from-createsignedurl)

**Reply by megacherry** [reply]
> @megacherry
> 
> If you have the path to the storage object, then it is pretty straight forward. You can do something like this to get the signed URL:
> 
> ```ts
> const pathToObject = ''; // Get your path to the object from DB
> const { signedURL, error } = await supabase
>   .storage
>   .from('avatars')
>   .createSignedUrl(pathToObject, 60)
> ```
> 
> [createSignedURL reference](https://supabase.io/docs/reference/javascript/storage-from-createsignedurl)

thanks for your answer. but that's the way i do it right now. i want to create the signed url inside my database function not in my javascript code. is that possible? :)

**Reply by dshukertjr** [reply]
@megacherry What is the end goal here? In the big picture what are you trying to achieve? There might be a better way of solving your problem.

**Reply by megacherry** [reply]
i only wanted to know if there is a way to generate the signed url on database side and give it right back to my application? or does that make no sense? and i have to handle it only client side? :-)

**Reply by dshukertjr** [reply]
@megacherry 
Creating a signed URL requires calling an endpoint, so you cannot truly do it within databases, but one thing you might be able to do is to setup [Function Hooks](https://supabase.io/blog/2021/07/30/supabase-functions-updates#function-hooks-alpha) to your database so that whenever a new data is inserted, you can call an external backend function where you will perform creating and saving the signed URL! 

 I think the most simple way of creating a signed URL is from the client, so not sure why you are avoiding that, but this way you won't have to do anything on the client side. Is this close to what you want to do?

**Reply by megacherry** [reply]
yes that was my intention. thank you for your detailed answer. :) i will try some things, we can mark it as answered. thank you very much for your time and help.

**Comment by megacherry** [comment]
Hi @dshukertjr

yes, i have the path to the storage object in database and the storage object information itself. All my business logic is stored in functions and i would like to generate the url directly on database side. Is this possible?
